lib/ansible/modules/cloud/amazon/route53_zone.py

```diff
@@ -195,31 +195,33 @@ def create_or_update_private(module, client, matching_zones, record):
         if record['vpc_id'] == current_vpc_id and record['vpc_region'] == current_vpc_region:
             record['zone_id'] = zone_details['Id'].replace('/hostedzone/', '')
             if 'Comment' in zone_details['Config'] and zone_details['Config']['Comment'] != record['comment']:
-                try:
-                    client.update_hosted_zone_comment(Id=zone_details['Id'], Comment=record['comment'])
-                except (BotoCoreError, ClientError) as e:
-                    module.fail_json_aws(e, msg="Could not update comment for hosted zone %s" % zone_details['Id'])
+                if not module.check_mode:
+                    try:
+                        client.update_hosted_zone_comment(Id=zone_details['Id'], Comment=record['comment'])
+                    except (BotoCoreError, ClientError) as e:
+                        module.fail_json_aws(e, msg="Could not update comment for hosted zone %s" % zone_details['Id'])
                 return True, record
             else:
                 record['msg'] = "There is already a private hosted zone in the same region with the same VPC \
                     you chose. Unable to create a new private hosted zone in the same name space."
                 return False, record
 
-    try:
-        result = client.create_hosted_zone(
-            Name=record['name'],
-            HostedZoneConfig={
-                'Comment': record['comment'] if record['comment'] is not None else "",
-                'PrivateZone': True,
-            },
-            VPC={
-                'VPCRegion': record['vpc_region'],
-                'VPCId': record['vpc_id'],
-            },
-            CallerReference="%s-%s" % (record['name'], time.time()),
-        )
-    except (BotoCoreError, ClientError) as e:
-        module.fail_json_aws(e, msg="Could not create hosted zone")
+    if not module.check_mode:
+        try:
+            result = client.create_hosted_zone(
+                Name=record['name'],
+                HostedZoneConfig={
+                    'Comment': record['comment'] if record['comment'] is not None else "",
+                    'PrivateZone': True,
+                },
+                VPC={
+                    'VPCRegion': record['vpc_region'],
+                    'VPCId': record['vpc_id'],
+                },
+                CallerReference="%s-%s" % (record['name'], time.time()),
+            )
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e, msg="Could not create hosted zone")
     hosted_zone = result['HostedZone']
```

Comment:
I think `result` isn't initialized when `module.check_mode` is `True`

Reply:
Fixed. Thank you for the review